auth: Make sure that we use strict weak records ordering in the API

rgacogne opened the pull request:
### Short description
<!-- Write a small description of what this Pull Request fixes or provides, including the issue #s -->

### Checklist
<!-- please indicate if any of these things are done/included with this Pull Request. Not all boxes need to be checked for the Pull Request to be accepted -->
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

zeha commented:
Please (manually) check that this doesn't break returning records+comments when there is an RRset with no records but comments left...

Apart from that LGTM

rgacogne commented:
> Please (manually) check that this doesn't break returning records+comments when there is an RRset with no records but comments left...

Nice catch, I reverted the ordering of records and comments in `fillZone()` to the existing behavior, and I added a test detecting a mismatch between records and comments.

rgacogne commented:
For the record, this solves an issue reported to the mailing-list: https://mailman.powerdns.com/pipermail/pdns-users/2018-July/025420.html
